lib/api/src/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2289,6 +2290,8 @@ postTransactionOld ctx@ApiLayer{..} argGenChange (ApiT wid) body = do
     let outs = addressAmountToTxOut <$> body ^. #payments
     let md = body ^? #metadata . traverse . #txMetadataWithSchema_metadata
     let mTTL = body ^? #timeToLive . traverse . #getQuantity
+    (Write.PParamsInAnyRecentEra era _, _)
+        <- liftIO $ W.readNodeTipStateForTxWrite netLayer
```

Comment:
Just noting: In general the idea with `readNodeTipStateForTxWrite` is that we only call it once during tx construction, to ensure a single source of truth and absence of race conditions. But here and now, even though there's another call in `W.buildSignSubmitTransaction`, I don't think we care too much.

Reply:
indeed, I wanted to cut it on server api level :+1: